### PR TITLE
First version

### PR DESCRIPTION
**Pull request by nuxlli**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
-# journey
+# Journey
+
 A simple sagas pattern implementation in Elixir
```
Comment (piacsek) [inline README.md]:
```suggestion
A simple [sagas](https://microservices.io/patterns/data/saga.html) pattern implementation in Elixir
```

File: lib/journey.ex [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    func.() |> extract_funcs()
+  end
+
+  defp extract_funcs({transaction, compensation} = funcs)
+       when is_valid_function(transaction) and is_valid_function(compensation),
+       do: funcs
+
+  defp extract_funcs(transaction) when is_valid_function(transaction), do: {transaction, nil}
+
+  defp run_compensation?(%__MODULE__{steps: steps} = journey) do
```
Comment (piacsek) [inline lib/journey.ex]: What about no `?` in the end of this function's name? It does not return a boolean.

File: lib/journey.ex [hunk 3]
```diff
@@ -0,0 +1,145 @@
+defmodule Journey.Step do
```
Comment (piacsek) [inline lib/journey.ex]: Since it's a lib, it'd be nice to have `@moduledoc` for modules and `@doc` for public functions.

Comment (nuxlli) [reply lib/journey.ex]: In the future ;)

**Review by piacsek — APPROVED** [review]
Good job, @nuxlli!

**Pull request merged**